ddtrace/llmobs/_integrations/openai_agents.py

```diff
@@ -296,3 +298,160 @@ def _llmobs_get_trace_info(
     def clear_state(self) -> None:
         self.oai_to_llmobs_span.clear()
         self.llmobs_traces.clear()
+
+    def tag_agent_manifest(self, span: Span, args: List[Any], kwargs: Dict[str, Any], agent_index: int) -> None:
+        agent = get_argument_value(args, kwargs, agent_index, "agent", None)
+        if not agent or not self.llmobs_enabled:
+            return
+        agent_manifest = self._create_agent_manifest(agent)
+        span._set_ctx_item(AGENT_MANIFEST, agent_manifest)
+
+    def _create_agent_manifest(self, agent):
+        manifest = {}
+        manifest["framework"] = "OpenAI"
+
+        if hasattr(agent, "name"):
+            manifest["name"] = agent.name
+        if hasattr(agent, "instructions"):
+            manifest["instructions"] = agent.instructions
+        if hasattr(agent, "handoff_description"):
+            manifest["handoff_description"] = agent.handoff_description
+        if hasattr(agent, "model"):
+            model = agent.model
+            manifest["model"] = model if isinstance(model, str) else getattr(model, "model", "")
+
+        model_settings = self._extract_model_settings_from_agent(agent)
+        if model_settings:
+            manifest["model_settings"] = model_settings
+
+        tools = self._extract_tools_from_agent(agent)
+        if tools:
+            manifest["tools"] = tools
+
+        handoffs = self._extract_handoffs_from_agent(agent)
+        if handoffs:
+            manifest["handoffs"] = handoffs
+
+        guardrails = self._extract_guardrails_from_agent(agent)
+        if guardrails:
+            manifest["guardrails"] = guardrails
+
+        return manifest
+
+    def _extract_model_settings_from_agent(self, agent):
+        if not hasattr(agent, "model_settings"):
+            return None
+
+        # convert model_settings to dict if it's not already
+        model_settings = agent.model_settings
+        if type(model_settings) != dict:
+            if hasattr(model_settings, "__dict__"):
+                model_settings = model_settings.__dict__
+            else:
+                return None
+
+        return self._make_json_compatible(model_settings)
+
+    def _extract_tools_from_agent(self, agent):
+        try:
+            from agents import (
+                WebSearchTool,
+                FileSearchTool,
+                ComputerTool,
+            )
+        except ImportError:
+            return None
```

Comment:
nice! i think maybe organizationally it makes sense to have this at the top of the file, but tbh i'm not too sure of the performance impacts either way, but thanks for try/catching it 😎